### ISSUE TITLE
isSpawnBiome - not working

### ISSUE THREAD

**Issue by Sir-Will** [superckl/BiomeTweaker#63]
I have set isSpawnBiome to false but I still frequently spawn in the ocean = 0
```
playerSpawn = forBiomes(0, 16, 24, 25, 59)
playerSpawn.set("isSpawnBiome", "false")
```

**Comment by superckl** [comment]
If it fails to find a viable spawn biome, it will spawn you there anyway. It will say in the log if it failed to do so. You'll have to add more biomes to possible spawn biomes to fix that.

**Comment by SirWilli** [comment]
Okay, I will check the log when it happens again.

**Comment by superckl** [comment]
The spawning code is very finicky. I'll take a closer look at it.

**Comment by superckl** [comment]
Search your logs for "Unable to find spawn biome". That's what MC will output when it fails to find a viable biome. You could also try applying the tweak around "PRE_INIT" and see if that helps. It can't hurt.

**Comment by SirWilli** [comment]
Yep, found it: https://gist.github.com/SirWilli/c5393e90dddf238e0cf3#file-biometweaker_log-L2395

What do you mean with apply a tweak around "PRE_INIT"?

**Comment by superckl** [comment]
So, since you found the log, that means you don't have enough spawn biomes, unfortunately. Ignore the "PRE_INIT" thing, not applicable anymore. I was talking about [this](https://github.com/superckl/BiomeTweaker/wiki/Basic-Script-Commands#setstage) command. It controls exactly when tweaks are applied during startup.

**Comment by superckl** [comment]
There's no way to "fix" this besides just adding more spawn biomes, so I'm closing this issue. We can continue discussing this here if needed.

**Comment by SirWilli** [comment]
Hm, the weird thing is I only remove it from few water biomes. So it should have enough spawn biomes.

**Comment by superckl** [comment]
Well, there are only 7 spawn biomes by default

```
forest, plains, taiga, taigaHills, forestHills, jungle, jungleHills
```

If you remove one or two of those, you're going to start seeing issue if you don't add more.

**Comment by SirWilli** [comment]
That's the current config: https://gist.github.com/SirWilli/6a988e611334c5952fb9
I added the spawn to all biomes except to the water biomes.

**Comment by superckl** [comment]
This build will output if a biome is a spawn biome to the output files:

http://jenkins.superckl.me/job/BiomeTweaker/180/

Is your world predominantly ocean? If so, it's possible to just get unlucky and spawn in the middle of one.

**Comment by SirWilli** [comment]
Sorry for the late reply. 
Looks like something is not working correctly with the spawn biome. Both biome files have `Spawn Biome: false` but one does have set `isSpawnBiome` to true and the other one false.

**Comment by SirWilli** [comment]
Nvm, found the issue. I had `"` around the true/false, after removing it the output file shows true but somehow nothing changed. I still spawn to 90% in water and get `Unable to find spawn biome`.
I think the option doesn't work.

**Comment by superckl** [comment]
The output files serve as a redundancy check, and show what a different mod (and Minecraft) would conclude with no knowledge of BiomeTweaker being present. If those say yes, than the answer is yes.

The spawning eventually looks for grass, yes, but not this part. The biome check is done before that, and the log you see is done when the biome check fails. I'm really not sure what else could be the issue. Could you try setting the application stage to "PRE_INIT"? e.g.

```
Tweaker.setStage("PRE_INIT")
playerSpawn = forBiomes(0, 16, 24, 25, 59)
playerSpawn.set("isSpawnBiome", "false")
Tweaker.setStage("FINISHED_LOAD")
```

**Comment by SirWilli** [comment]
You mean like this? https://gist.github.com/SirWilli/f9acf14d2b8e3bb16d40
Doesn't seem to change anything

**Comment by SirWilli** [comment]
@superckl any news to this?

**Comment by superckl** [comment]
Nope, still clueless.

**Comment by SirWilli** [comment]
The weird thing is, if I add ClimateControl the spawning works fine.

**Comment by SirWilli** [comment]
Maybe @Zeno410 does have a idea.

**Comment by superckl** [comment]
Well, I'm heavily considering dropping development on the 1.7.10 branch. No guarantees I will fix this unless it's really easy or still an issue in 1.8. Since I have no clue why this is happening, I don't think it's an easy fix.

**Comment by superckl** [comment]
Can you confirm this is still an issue in 1.8+? If not, this will stay closed since development has stopped on 1.7.10.
